test/battle/move_effect/burn_hit.c

```diff
@@ -38,3 +38,32 @@ SINGLE_BATTLE_TEST("Ember cannot burn a Fire-type Pokémon")
         }
     }
 }
+
+#if B_STATUS_TYPE_IMMUNITY > GEN_1
+#define B_STATUS_TYPE_IMMUNITY_TEST "Scald should burn a Water-type Pokémon"
+#else
+#define B_STATUS_TYPE_IMMUNITY_TEST "Scald shouldn't burn a Water-type Pokémon"
+#endif
+SINGLE_BATTLE_TEST(B_STATUS_TYPE_IMMUNITY_TEST)
```

Comment:
I think you may be able to write this a little shorter (and you could drop the `#undef`):
```suggestion
#if B_STATUS_TYPE_IMMUNITY > GEN_1
SINGLE_BATTLE_TEST("Scald should burn a Water-type Pokémon")
#else
SINGLE_BATTLE_TEST("Scald shouldn't burn a Water-type Pokémon")
#endif
```

Reply:
Thanks, this was reworded.